ui/App.tsx

```diff
@@ -65,41 +67,60 @@ function withSearchParams(Cmp) {
   };
 }
 
-const navItems: NavItem[] = [
-  {
-    label: "Applications",
-    link: { value: V2Routes.Automations },
-    icon: IconType.ApplicationsIcon,
-  },
-  {
-    label: "Sources",
-    link: { value: V2Routes.Sources },
-    icon: IconType.SourcesIcon,
-  },
-  {
-    label: "Image Automation",
-    link: { value: V2Routes.ImageAutomation },
-    icon: IconType.ImageAutomationIcon,
-  },
-  {
-    label: "Policies",
-    link: { value: V2Routes.Policies },
-    icon: IconType.PoliciesIcon,
-  },
-  {
+function getRuntimeNavItem(isNewRuntimeEnabled: boolean): NavItem {
+  if (isNewRuntimeEnabled) {
+    return {
+      label: "Runtime",
+      link: { value: V2Routes.Runtime },
+      icon: IconType.FluxIcon,
+    };
+  }
+
+  return {
     label: "Flux Runtime",
     link: { value: V2Routes.FluxRuntime },
     icon: IconType.FluxIcon,
-  },
-  {
-    label: "Notifications",
-    link: { value: V2Routes.Notifications },
-    icon: IconType.NotificationsIcon,
-  },
-];
+  };
+}
 
 const App = () => {
   const dark = useInDarkMode();
+
+  const { isFlagEnabled } = useFeatureFlags();
+
+  const isNewRuntimeEnabled = isFlagEnabled(
+    "WEAVE_GITOPS_FEATURE_GITOPS_RUNTIME"
+  );
+
+  const navItems: NavItem[] = [
```

Comment:
could you expand here, not sure if fully understand.

Reply:
In main the `NavItems` constant was declared outside of the App function component:
https://github.com/weaveworks/weave-gitops/blob/78d5f6dc24cbc647800bfeed3691c80c517d2edf/ui/App.tsx#L68

Here it is moved to within the App function/component.

You can just move the first four items to where the whole array was like this at line 70, like:

```
const navItemsStart: NavItem[] = [
  {
    label: "Applications",
    link: { value: V2Routes.Automations },
    icon: IconType.ApplicationsIcon,
  },
  {
    label: "Sources",
    link: { value: V2Routes.Sources },
    icon: IconType.SourcesIcon,
  },
  {
    label: "Image Automation",
    link: { value: V2Routes.ImageAutomation },
    icon: IconType.ImageAutomationIcon,
  },
  {
    label: "Policies",
    link: { value: V2Routes.Policies },
    icon: IconType.PoliciesIcon,
  }];
```

And then instead of the whole const NavItems array in the App function just combine the static and dynamic parts there (at line 95):

```
const navItems: NavItem[] = [
    ...navItemsStart,
    getRuntimeNavItem(isNewRuntimeEnabled),
    {
      label: "Notifications",
      link: { value: V2Routes.Notifications },
      icon: IconType.NotificationsIcon,
    },
  ];
```

Again, this is an optional suggestion, if you prefer to keep the whole array in the App function for readability, go with it.

Reply:
got it now, yes, i think i leave it like this  🙏